File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {

        unordered_map<int,int> frequency_map;

        for(auto num : nums){
            frequency_map[num]++;
        }

        int maxFreq = 0;
        int keywithMax;

        for(auto pair : frequency_map){
            if(pair.second > maxFreq) {
                maxFreq = pair.second;
                keywithMax = pair.first;
            }
        }
    return keywithMax;
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int n = nums.length;

        for(int x : nums){
            map.put(x,map.getOrDefault(x,0) + 1);
        }

        for(Map.Entry<Integer,Integer> e : map.entrySet()){
            if(e.getValue() > (n/2)){
                return e.getKey();
            }
        }
    return -1;
    }
}